### PR TITLE
[Hexagon] Add test demonstrating how to utilize a rolling buffer for avoiding repeated loads

### PR DESCRIPTION
**Pull request by csullivan**
This PR shows how to utilize the `sch.rolling_buffer` schedule primitive to avoid copying repeated rows when loading to fast memory. The example below is like conv2d with filter=[2, 1] and unit weights.

Before applying a rolling buffer to the height axis: 
```python

# from tvm.script import tir as T
@tvm.script.ir_module
class Module:
    @T.prim_func
    def main(A: T.Buffer[(6, 32), "float16"], B: T.Buffer[(5, 32), "float16"]):
        # function attr dict
        T.func_attr({"global_symbol": "main"})
        # body
        # with T.block("root")
        A_global_vtcm = T.alloc_buffer([6, 32], dtype="float16", scope="global.vtcm")
        for output_row in T.serial(5):
            for ax0, ax1 in T.grid(2, 32):
                with T.block("A_global.vtcm"):
                    v0 = T.axis.spatial(6, output_row + ax0)
                    v1 = T.axis.spatial(32, ax1)
                    T.reads(A[v0, v1])
                    T.writes(A_global_vtcm[v0, v1])
                    A_global_vtcm[v0, v1] = A[v0, v1]
            for i, rh in T.grid(32, 2):
                with T.block("compute"):
                    vrow, vi, vrh = T.axis.remap("SSR", [output_row, i, rh])
                    T.reads(A_global_vtcm[vrow + vrh, vi])
                    T.writes(B[vrow, vi])
                    with T.init():
                        B[vrow, vi] = T.cast(0, "float16")
                    B[vrow, vi] = B[vrow, vi] + A_global_vtcm[vrow + vrh, vi]

```
After applying `sch.rolling_buffer` to the cache_read block along the height axis:

```python
# from tvm.script import tir as T
@tvm.script.ir_module
class Module:
    @T.prim_func
    def main(A: T.Buffer[(6, 32), "float16"], B: T.Buffer[(5, 32), "float16"]):
        # function attr dict
        T.func_attr({"global_symbol": "main"})
        # body
        # with T.block("root")
        A_global_vtcm = T.alloc_buffer([2, 32], dtype="float16", scope="global.vtcm")
        for output_row in T.serial(5):
            for ax0, ax1 in T.grid(2, 32):
                with T.block("A_global.vtcm"):
                    T.where(output_row < 1 or 1 <= ax0)
                    v0 = T.axis.opaque(6, output_row + ax0)
                    v1 = T.axis.spatial(32, ax1)
                    T.reads(A[v0, v1])
                    T.writes(A_global_vtcm[v0 % 2, v1])
                    A_global_vtcm[v0 % 2, v1] = A[v0, v1]
            for i, rh in T.grid(32, 2):
                with T.block("compute"):
                    vrow = T.axis.opaque(5, output_row)
                    vi, vrh = T.axis.remap("SR", [i, rh])
                    T.reads(A_global_vtcm[(vrow + vrh) % 2, vi])
                    T.writes(B[vrow, vi])
                    with T.init():
                        B[vrow, vi] = T.cast(0, "float16")
                    B[vrow, vi] = B[vrow, vi] + A_global_vtcm[(vrow + vrh) % 2, vi]

```

### CODE REVIEW

**Comment by csullivan** [comment]
cc @kparzysz-quic

Many thanks (!!) to @mbaret @NicolaLancellotti for the initial contribution of rolling buffer [[1](https://discuss.tvm.apache.org/t/rfc-introducing-a-rolling-buffer-scheduling-primitive/9836), [2](https://github.com/apache/tvm/pull/9444)], and to @liangW-intellif and @wrongtest-intellif for their recent translation of rolling buffer to a TIR schedule primitive [[3](https://github.com/apache/tvm/pull/13033)], as well as all others involved in this work. It is a fantastic example of the benefit of a strong open source community!

**Review by masahi — APPROVED** [review]

File: tests/python/contrib/test_hexagon/test_rolling_buffer.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    vcol = T.axis.spatial(32, i)
+                    vrh = T.axis.reduce(2, reduce_h)
+                    with T.init():
+                        b_buffer[vrow, vcol] = T.cast(0, dtype=dtype)
+                    b_buffer[vrow, vcol] = b_buffer[vrow, vcol] + a_buffer[vrow + vrh, vcol]
+
+    return conv2d_2x1_filter
+
+
+def test_rolling_buffer_conv2d_2x1(hexagon_session):
```
Comment (mehrdadh) [inline tests/python/contrib/test_hexagon/test_rolling_buffer.py:49]: Please add `@tvm.testing.requires_hexagon`

File: tests/python/contrib/test_hexagon/test_rolling_buffer.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    sch.compute_at(cache_read_block, output_row)
+    sch.rolling_buffer(cache_read_block, 0)
+
+    target = get_hexagon_target("v69")
+    mod = tvm.build(sch.mod, target=target)
+    mod = hexagon_session.load_module(mod)
+    a = tvm.nd.array(np.ones((6, 32), dtype=dtype), device=hexagon_session.device)
+    b = tvm.nd.array(np.zeros((5, 32), dtype=dtype), device=hexagon_session.device)
+    mod(a, b)
+    tvm.testing.assert_allclose(b.numpy(), np.full((5, 32), 2, dtype=dtype), atol=1e-3, rtol=1e-3)
```
Comment (mehrdadh) [inline tests/python/contrib/test_hexagon/test_rolling_buffer.py:65]: Please add this to the end:

```
if __name__ == "__main__":
    tvm.testing.main()
```

**Review by mehrdadh — CHANGES_REQUESTED** [review]
LGTM! just small changes

**Comment by csullivan** [comment]
@tvm-bot rerun

**Comment by masahi** [comment]
@tvm-bot rerun

**Comment by kparzysz-quic** [comment]
@tvm-bot rerun

**Comment by csullivan** [comment]
@tvm-bot rerun

**Pull request closed without merging**